lxd/images.go

```diff
@@ -793,7 +793,16 @@ func imagesPost(d *Daemon, r *http.Request) response.Response {
 		return nil
 	}
 
-	op, err := operations.OperationCreate(d.State(), project, operations.OperationClassTask, db.OperationImageDownload, nil, nil, run, nil, nil)
+	var metadata interface{}
+
+	secret := r.Header.Get("X-LXD-secret")
+	if secret != "" {
+		metadata = map[string]string{
+			"secret": secret,
+		}
+	}
```

Comment:
Not sure what's that about or if still relevant with the changes to use ?secret= on operations.

Reply:
Well, the part that seems odd is that we're recycling a secret. I think I'd prefer we generate a new one. The caller will get the secret through the initial operation struct it receives, then can call wait on that.

Reply:
We could also generate two secret, passing both to the source. One of the secrets can then be used to call `ImageCreate`, the other one on wait. That might look a bit cleaner.

Reply:
I'm not sure why we'd do that.

Secrets should be generated when an operation is created and will be accessible to the caller in the struct they receive as a result.

So in this case:
 - CLI calls destination, telling about incoming push image, gets a token secret from that op
 - CLI calls source, gives destination details and token secret
 - Source pushes image to the destination using the token secret, new operation is created to process the image and has its own secret
 - Source waits on that operation using the secret it received during push

Reply:
That means, instead of doing `secret := r.Header.Get("X-LXD-secret")` we just do `secret := <generate new secret>`.

Reply:
yes